infrastructure/smart-contracts/contracts/upgrades/node-license/NodeLicense8.sol

```diff
@@ -288,6 +298,38 @@ contract NodeLicense8 is ERC721EnumerableUpgradeable, AccessControlUpgradeable
         }     
     }
 
+    /**
+     * @dev Mints a node license to the recipient address where the caller pays in USDC.
+     */
+    function mintToWithUSDC(address _to, uint256 _amount, string calldata _promoCode, uint256 _expectedCostInUSDC) public {
+        //validate
+        _validateMint(_amount);
+
+        //convert the final price to USDC
+        uint256 mintPriceInWei = price(_amount, _promoCode);
+        uint256 ethPriceInUSDC = uint256(ethPriceFeed.latestAnswer()) * 10**10; //convert to 18 decimals
+        uint256 mintPriceInUSDC = (mintPriceInWei * ethPriceInUSDC) / 10**18;
+        require(mintPriceInUSDC <= _expectedCostInUSDC, "Price Exceeds Expected Cost");
+
+        //mint node license tokens
+        uint256 averageCost = mintPriceInWei / _amount;
+        _mintNodeLicense(_amount, averageCost, _to);
+
+        //calculate referral reward and determine if the promo code is an address
+        (uint256 referralReward, address recipientAddress) = _calculateReferralReward(mintPriceInUSDC, _promoCode);
+
+        //transfer usdc from caller
+        IERC20 token = IERC20(usdcAddress);
+        token.transferFrom(msg.sender, address(this), mintPriceInUSDC);
+        token.transfer(fundsReceiver, mintPriceInUSDC - referralReward);
+
+        if (referralReward > 0) {
+            //store the referral reward in the appropriate mapping
+            _promoCodesXai[_promoCode].receivedLifetime += referralReward;
+            _referralRewardsXai[recipientAddress] += referralReward;
```

Comment:
We will have to handle that with its own state or we won't know what was Xai and what was USDC

Reply:
We will also have to be able to claim the USDC referral amounts. THis should have been part of the story description or we need to create a new ticket for that